zanata-war/src/main/java/org/zanata/adapter/OkapiFilterAdapter.java

```diff
@@ -172,6 +168,9 @@ public Resource parseDocumentFile(URI documentContent,
         RawDocument rawDoc =
                 new RawDocument(documentContent, "UTF-8",
                         net.sf.okapi.common.LocaleId.fromString("en"));
+        if (rawDoc.getTargetLocale() == null) {
+            rawDoc.setTargetLocale(net.sf.okapi.common.LocaleId.ENGLISH);
```

Comment:
@seanf so, in pushing a source file, the targetLocale should not be relevant - yes?
It causes a nullpointer exception if it's not set for TS - maybe it's a bug?